subsys/llext/llext_link.c

```diff
@@ -465,6 +465,20 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, const struct llext_l
 			sys_cache_instr_invd_range(ext->mem[i], ext->mem_size[i]);
 		}
 	}
+
+	/* Detached section caches should be synchronized in place */
+	if (ldr_parm->section_detached) {
+		for (i = 0; i < ldr->sect_cnt; ++i) {
+			elf_shdr_t *shdr = ldr->sect_hdrs + i;
+
+			if (ldr_parm->section_detached(shdr)) {
```

Comment:
Should this be wrapped in a compile time branch? E.g.
```c
if(IS_ENABLED(CONFIG_CACHE_MANAGEMENT) && ldr_param->section_detached) {
}
```

Otherwise we're doing more work than needed

Reply:
> Should this be wrapped in a compile time branch? E.g.
> 
> ```c
> if(IS_ENABLED(CONFIG_CACHE_MANAGEMENT) && ldr_param->section_detached) {
> }
> ```
> 
> Otherwise we're doing more work than needed

@teburd there's a check for `ldr_parm->section_detached != NULL` above the loop, I don't think anyone will have that function supplied to LLEXT if they don't want to use it, but yes, we can add the check there too

Reply:
Ah the github diff didn't show that much, thanks!